nano/node/node.cpp

```diff
@@ -1120,72 +1117,152 @@ nano::signature_checker::~signature_checker ()
 	stop ();
 }
 
-void nano::signature_checker::add (nano::signature_check_set & check_a)
+void nano::signature_checker::verify (nano::signature_check_set & check_a)
 {
 	{
-		std::lock_guard<std::mutex> lock (mutex);
-		checks.push_back (check_a);
+		// Don't process anything else if we have stopped
+		std::lock_guard<std::mutex> guard (stopped_mutex);
+		if (stopped)
+		{
+			return;
+		}
 	}
-	condition.notify_all ();
+
+	if (check_a.size < multithreaded_cutoff || single_threaded)
+	{
+		// Not dealing with many signatures so just use the calling thread for checking signatrure
+		verify_batch (check_a, 0, check_a.size);
+		return;
```

Comment:
Missing a release_assert on results of this verify_batch

Reply:
Good spot, thanks!